subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/LegacyTransformer.java

```diff
@@ -51,7 +51,17 @@ public boolean requiresDependencies() {
         if (outputs == null) {
             throw new InvalidUserDataException("Transform returned null result.");
         }
-        return validateOutputs(primaryInput, outputDir, ImmutableList.copyOf(outputs));
+        validateOutputs(primaryInput, outputDir, outputs);
+        return ImmutableList.copyOf(outputs);
+    }
+
+    private static void validateOutputs(File primaryInput, File outputDir, List<File> outputs) {
+        String inputFilePrefix = primaryInput.getPath() + File.separator;
+        String outputDirPrefix = outputDir.getPath() + File.separator;
+        for (File output : outputs) {
+            DefaultArtifactTransformOutputs.validateOutputExists(output);
+            DefaultArtifactTransformOutputs.validateOutput(output, primaryInput, inputFilePrefix, outputDir, outputDirPrefix);
```

Comment:
This looks like calling into the implementation. If this functionality is shared, it should not be in a `Default` class I think.

Reply:
I moved the implementation to `ArtifactTransformOutputsInternal`.